### PR TITLE
Ignore single Json record if size is larger than 256KB which is size limit for eventhub

### PR DESCRIPTION
**Pull request by yukuai518**
Tolerance is chosen to be 95% so the internal size limit will be (256K * 0.95)

### CODE REVIEW

File: gobblin-modules/gobblin-eventhub/src/main/java/gobblin/eventhub/writer/EventhubBatch.java [hunk 1]
```diff
@@ -39,6 +39,7 @@
   private final long creationTimestamp;
   private final long memSizeLimit;
   private final long ttlInMilliSeconds;
+  public static final int overheadSizeInByte = 15;
```
Comment (ibuenros) [inline gobblin-modules/gobblin-eventhub/src/main/java/gobblin/eventhub/writer/EventhubBatch.java]: `static final` variables should have names in all caps `OVERHEAD_SIZE_IN_BYTES`

File: gobblin-modules/gobblin-eventhub/src/main/java/gobblin/eventhub/writer/EventhubBatchAccumulator.java [hunk 2]
```diff
@@ -85,17 +93,27 @@ public long getExpireInMilliSecond () {
...
-      EventhubBatch batch = new EventhubBatch(this.batchSizeLimit, this.expireInMilliSecond);
+      EventhubBatch batch = new EventhubBatch(this.memSizeLimit, this.expireInMilliSecond);
       LOG.info ("Batch " + batch.getId() + " is generated");
       Future<RecordMetadata> future = batch.tryAppend(record, callback);
+
+      // Even single record can exceed the size limit from one batch
+      // Ignore the record because Eventhub can only accept payload less than 256KB
+      if (future == null) {
+        LOG.warn ("Batch " + batch.getId() + " is marked as complete because it contains a huge record: " + record.toString());
+        future = new RecordFuture(new CountDownLatch(0), 0);
```
Comment (ibuenros) [inline gobblin-modules/gobblin-eventhub/src/main/java/gobblin/eventhub/writer/EventhubBatchAccumulator.java]: You can use an `ImmediateFailedFuture` here (or `ImmeduateSuccessfulFuture`)

Comment (yukuai518) [inline gobblin-modules/gobblin-eventhub/src/main/java/gobblin/eventhub/writer/EventhubBatchAccumulator.java]: Done

File: gobblin-modules/gobblin-eventhub/src/main/java/gobblin/eventhub/writer/EventhubBatch.java (hunk 1 above)
Comment (yukuai518) [inline gobblin-modules/gobblin-eventhub/src/main/java/gobblin/eventhub/writer/EventhubBatch.java]: Done

**Comment by yukuai518** [comment]
Test failed in gobblin-cluster module which is not related to this commit.

**Pull request merged**